scripts/generate-config-data.js

```diff
@@ -0,0 +1,245 @@
+const https = require('https');
+const fs = require('fs');
+const path = require('path');
+
+function fetchSource () {
+  return new Promise(resolve => {
+    https.get('https://raw.githubusercontent.com/paritytech/parity/master/parity/cli/mod.rs', res => {
+      res.setEncoding('utf8');
+      let body = '';
+      res.on('data', data => { body += data; });
```

Comment:
Could use https://github.com/rvagg/bl and `require('util').promisify` to simplify processing.

Reply:
cool! didn't know about `bl`

However I ended up with the following structure and I couldn't see how I could fit `promisify` in there to further simplify:

```javascript
function fetchSource () {
  return new Promise((resolve, reject) => {
    https.get('https://raw.githubusercontent.com/paritytech/parity/master/parity/cli/mod.rs', res => {
      if (res.statusCode !== 200) {
        reject(new Error(`Failed to load source code, status code: ${res.statusCode}`));
      }

      res.pipe(bl(function (err, data) {
        if (err) {
          reject(new Error(err));
        }

        resolve(data.toString());
      }))
    });
  });
}
```